private/model/api/docstring.go

```diff
@@ -88,21 +92,134 @@ func docstring(doc string) string {
 	doc = reComments.ReplaceAllString(doc, "")
 	doc = reFullname.ReplaceAllString(doc, "")
 	doc = reExamples.ReplaceAllString(doc, "")
-	doc = rePara.ReplaceAllString(doc, "$1\n\n")
-	doc = reLink.ReplaceAllString(doc, "$2 ($1)")
-	doc = reTag.ReplaceAllString(doc, "$1")
+
+	_, err := xhtml.Parse(strings.NewReader(doc))
+	// If there is no error, means it is HTML. However, if it isn't HTML, we use the old
+	// way of generating documentation.
+	if err == nil {
+		// this will only occur if an error during the tokenization process occurs.
+		doc = generateDocFromHTML(doc)
+	} else {
+		doc = rePara.ReplaceAllString(doc, "$1\n\n")
+		doc = reLink.ReplaceAllString(doc, "$2 ($1)")
+	}
+
 	doc = reEndNL.ReplaceAllString(doc, "")
-	doc = strings.TrimSpace(doc)
 	if doc == "" {
 		return "\n"
 	}
 
 	doc = html.UnescapeString(doc)
-	doc = wrap(doc, 72)
-
 	return commentify(doc)
 }
 
+// generateDocFromHTML will generate the proper doc string for html encoded doc entries.
+func generateDocFromHTML(htmlSrc string) string {
```

Comment:
Yea, the reason why it does that is due to how this parser works. I am working on a modification to extend it to be smarter about nested tags.